// @azure/arm-hdinsight/esm/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
export var acceptLanguage = {
    parameterPath: "acceptLanguage",
    mapper: {
        serializedName: "accept-language",
        defaultValue: 'en-US',
        type: {
            name: "String"
        }
    }
};
export var apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        required: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export var applicationName = {
    parameterPath: "applicationName",
    mapper: {
        required: true,
        serializedName: "applicationName",
        type: {
            name: "String"
        }
    }
};
export var clusterName = {
    parameterPath: "clusterName",
    mapper: {
        required: true,
        serializedName: "clusterName",
        type: {
            name: "String"
        }
    }
};
export var configurationName = {
    parameterPath: "configurationName",
    mapper: {
        required: true,
        serializedName: "configurationName",
        type: {
            name: "String"
        }
    }
};
export var extensionName = {
    parameterPath: "extensionName",
    mapper: {
        required: true,
        serializedName: "extensionName",
        type: {
            name: "String"
        }
    }
};
export var location = {
    parameterPath: "location",
    mapper: {
        required: true,
        serializedName: "location",
        type: {
            name: "String"
        }
    }
};
export var nextPageLink = {
    parameterPath: "nextPageLink",
    mapper: {
        required: true,
        serializedName: "nextLink",
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export var resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        required: true,
        serializedName: "resourceGroupName",
        type: {
            name: "String"
        }
    }
};
export var roleName = {
    parameterPath: "roleName",
    mapper: {
        required: true,
        isConstant: true,
        serializedName: "roleName",
        defaultValue: 'workernode',
        type: {
            name: "String"
        }
    }
};
export var scriptExecutionId = {
    parameterPath: "scriptExecutionId",
    mapper: {
        required: true,
        serializedName: "scriptExecutionId",
        type: {
            name: "String"
        }
    }
};
export var scriptName = {
    parameterPath: "scriptName",
    mapper: {
        required: true,
        serializedName: "scriptName",
        type: {
            name: "String"
        }
    }
};
export var subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        required: true,
        serializedName: "subscriptionId",
        type: {
            name: "String"
        }
    }
};
