// @azure/arm-subscriptions/esm/models/mappers.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import { CloudErrorMapper, BaseResourceMapper } from "@azure/ms-rest-azure-js";
export var CloudError = CloudErrorMapper;
export var BaseResource = BaseResourceMapper;
export var SubscriptionCreationResult = {
    serializedName: "SubscriptionCreationResult",
    type: {
        name: "Composite",
        className: "SubscriptionCreationResult",
        modelProperties: {
            subscriptionLink: {
                serializedName: "subscriptionLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var AdPrincipal = {
    serializedName: "AdPrincipal",
    type: {
        name: "Composite",
        className: "AdPrincipal",
        modelProperties: {
            objectId: {
                required: true,
                serializedName: "objectId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var SubscriptionCreationParameters = {
    serializedName: "SubscriptionCreationParameters",
    type: {
        name: "Composite",
        className: "SubscriptionCreationParameters",
        modelProperties: {
            displayName: {
                serializedName: "displayName",
                type: {
                    name: "String"
                }
            },
            owners: {
                serializedName: "owners",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "AdPrincipal"
                        }
                    }
                }
            },
            offerType: {
                serializedName: "offerType",
                type: {
                    name: "String"
                }
            },
            additionalParameters: {
                serializedName: "additionalParameters",
                type: {
                    name: "Dictionary",
                    value: {
                        type: {
                            name: "Object"
                        }
                    }
                }
            }
        }
    }
};
export var ErrorResponse = {
    serializedName: "ErrorResponse",
    type: {
        name: "Composite",
        className: "ErrorResponse",
        modelProperties: {
            code: {
                serializedName: "code",
                type: {
                    name: "String"
                }
            },
            message: {
                serializedName: "message",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var SubscriptionOperation = {
    serializedName: "SubscriptionOperation",
    type: {
        name: "Composite",
        className: "SubscriptionOperation",
        modelProperties: {
            id: {
                readOnly: true,
                serializedName: "id",
                type: {
                    name: "String"
                }
            },
            status: {
                serializedName: "status",
                type: {
                    name: "String"
                }
            },
            statusDetail: {
                serializedName: "statusDetail",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var SubscriptionOperationListResult = {
    serializedName: "SubscriptionOperationListResult",
    type: {
        name: "Composite",
        className: "SubscriptionOperationListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "SubscriptionOperation"
                        }
                    }
                }
            }
        }
    }
};
export var OperationDisplay = {
    serializedName: "Operation_display",
    type: {
        name: "Composite",
        className: "OperationDisplay",
        modelProperties: {
            provider: {
                serializedName: "provider",
                type: {
                    name: "String"
                }
            },
            resource: {
                serializedName: "resource",
                type: {
                    name: "String"
                }
            },
            operation: {
                serializedName: "operation",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var Operation = {
    serializedName: "Operation",
    type: {
        name: "Composite",
        className: "Operation",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            display: {
                serializedName: "display",
                type: {
                    name: "Composite",
                    className: "OperationDisplay"
                }
            }
        }
    }
};
export var OperationListResult = {
    serializedName: "OperationListResult",
    type: {
        name: "Composite",
        className: "OperationListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Operation"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var Location = {
    serializedName: "Location",
    type: {
        name: "Composite",
        className: "Location",
        modelProperties: {
            id: {
                readOnly: true,
                serializedName: "id",
                type: {
                    name: "String"
                }
            },
            subscriptionId: {
                readOnly: true,
                serializedName: "subscriptionId",
                type: {
                    name: "String"
                }
            },
            name: {
                readOnly: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            displayName: {
                readOnly: true,
                serializedName: "displayName",
                type: {
                    name: "String"
                }
            },
            latitude: {
                readOnly: true,
                serializedName: "latitude",
                type: {
                    name: "String"
                }
            },
            longitude: {
                readOnly: true,
                serializedName: "longitude",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var SubscriptionPolicies = {
    serializedName: "SubscriptionPolicies",
    type: {
        name: "Composite",
        className: "SubscriptionPolicies",
        modelProperties: {
            locationPlacementId: {
                readOnly: true,
                serializedName: "locationPlacementId",
                type: {
                    name: "String"
                }
            },
            quotaId: {
                readOnly: true,
                serializedName: "quotaId",
                type: {
                    name: "String"
                }
            },
            spendingLimit: {
                readOnly: true,
                serializedName: "spendingLimit",
                type: {
                    name: "Enum",
                    allowedValues: [
                        "On",
                        "Off",
                        "CurrentPeriodOff"
                    ]
                }
            }
        }
    }
};
export var Subscription = {
    serializedName: "Subscription",
    type: {
        name: "Composite",
        className: "Subscription",
        modelProperties: {
            id: {
                readOnly: true,
                serializedName: "id",
                type: {
                    name: "String"
                }
            },
            subscriptionId: {
                readOnly: true,
                serializedName: "subscriptionId",
                type: {
                    name: "String"
                }
            },
            displayName: {
                readOnly: true,
                serializedName: "displayName",
                type: {
                    name: "String"
                }
            },
            state: {
                readOnly: true,
                serializedName: "state",
                type: {
                    name: "Enum",
                    allowedValues: [
                        "Enabled",
                        "Warned",
                        "PastDue",
                        "Disabled",
                        "Deleted"
                    ]
                }
            },
            subscriptionPolicies: {
                serializedName: "subscriptionPolicies",
                type: {
                    name: "Composite",
                    className: "SubscriptionPolicies"
                }
            },
            authorizationSource: {
                serializedName: "authorizationSource",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var TenantIdDescription = {
    serializedName: "TenantIdDescription",
    type: {
        name: "Composite",
        className: "TenantIdDescription",
        modelProperties: {
            id: {
                readOnly: true,
                serializedName: "id",
                type: {
                    name: "String"
                }
            },
            tenantId: {
                readOnly: true,
                serializedName: "tenantId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var SubscriptionFactoryCreateSubscriptionInEnrollmentAccountHeaders = {
    serializedName: "subscriptionfactory-createsubscriptioninenrollmentaccount-headers",
    type: {
        name: "Composite",
        className: "SubscriptionFactoryCreateSubscriptionInEnrollmentAccountHeaders",
        modelProperties: {
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            },
            retryAfter: {
                serializedName: "retry-after",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var LocationListResult = {
    serializedName: "LocationListResult",
    type: {
        name: "Composite",
        className: "LocationListResult",
        modelProperties: {
            value: {
                serializedName: "",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Location"
                        }
                    }
                }
            }
        }
    }
};
export var SubscriptionListResult = {
    serializedName: "SubscriptionListResult",
    type: {
        name: "Composite",
        className: "SubscriptionListResult",
        modelProperties: {
            value: {
                serializedName: "",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Subscription"
                        }
                    }
                }
            },
            nextLink: {
                required: true,
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var TenantListResult = {
    serializedName: "TenantListResult",
    type: {
        name: "Composite",
        className: "TenantListResult",
        modelProperties: {
            value: {
                serializedName: "",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "TenantIdDescription"
                        }
                    }
                }
            },
            nextLink: {
                required: true,
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
