// @azure/arm-hdinsight/esm/operations/locations.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/locationsMappers";
import * as Parameters from "../models/parameters";
/** Class representing a Locations. */
var Locations = /** @class */ (function () {
    /**
     * Create a Locations.
     * @param {HDInsightManagementClientContext} client Reference to the service client.
     */
    function Locations(client) {
        this.client = client;
    }
    Locations.prototype.listUsages = function (location, options, callback) {
        return this.client.sendOperationRequest({
            location: location,
            options: options
        }, listUsagesOperationSpec, callback);
    };
    return Locations;
}());
export { Locations };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var listUsagesOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.HDInsight/locations/{location}/usages",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.location
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.UsagesListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
