// @azure/arm-hdinsight/esm/operations/scriptActions.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/scriptActionsMappers";
import * as Parameters from "../models/parameters";
/** Class representing a ScriptActions. */
var ScriptActions = /** @class */ (function () {
    /**
     * Create a ScriptActions.
     * @param {HDInsightManagementClientContext} client Reference to the service client.
     */
    function ScriptActions(client) {
        this.client = client;
    }
    ScriptActions.prototype.deleteMethod = function (resourceGroupName, clusterName, scriptName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            clusterName: clusterName,
            scriptName: scriptName,
            options: options
        }, deleteMethodOperationSpec, callback);
    };
    ScriptActions.prototype.listByCluster = function (resourceGroupName, clusterName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            clusterName: clusterName,
            options: options
        }, listByClusterOperationSpec, callback);
    };
    ScriptActions.prototype.getExecutionDetail = function (resourceGroupName, clusterName, scriptExecutionId, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            clusterName: clusterName,
            scriptExecutionId: scriptExecutionId,
            options: options
        }, getExecutionDetailOperationSpec, callback);
    };
    ScriptActions.prototype.listByClusterNext = function (nextPageLink, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            options: options
        }, listByClusterNextOperationSpec, callback);
    };
    return ScriptActions;
}());
export { ScriptActions };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var deleteMethodOperationSpec = {
    httpMethod: "DELETE",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HDInsight/clusters/{clusterName}/scriptActions/{scriptName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.clusterName,
        Parameters.scriptName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {},
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
var listByClusterOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HDInsight/clusters/{clusterName}/scriptActions",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.clusterName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.ScriptActionsList
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
var getExecutionDetailOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HDInsight/clusters/{clusterName}/scriptExecutionHistory/{scriptExecutionId}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.clusterName,
        Parameters.scriptExecutionId
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.RuntimeScriptActionDetail
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
var listByClusterNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.ScriptActionsList
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
